tokio-macros/src/entry.rs

```diff
@@ -380,7 +380,64 @@ fn parse_knobs(mut input: ItemFn, is_test: bool, config: FinalConfig) -> TokenSt
         }
     };
 
-    let body = input.body();
+    let unit_type = Box::new(parse_quote! { () });
+    let never_type = Box::new(parse_quote! { ! });
+
+    let output_type = match &input.sig.output {
+        // For functions with no return value syn doesn't print anything,
+        // but that doesn't work as `Output` for our boxed `Future`, so
+        // default to `()` (the same type as the function output).
+        syn::ReturnType::Default => &unit_type,
+        syn::ReturnType::Type(_, ret_type) if ret_type == &never_type => &unit_type,
+        syn::ReturnType::Type(_, ret_type) => ret_type,
+    };
+
+    enum ContainsLoopVisitor {
+        Found,
+        NotFound,
+    }
+
+    impl syn::visit::Visit<'_> for ContainsLoopVisitor {
```

Comment:
When you say the "return value," do you mean the last statement? I guess any statement that contained a `loop` that doesn't exit would have to be the last statement otherwise any statements that follow would be dead code.

So that would be an improvement, but that doesn't change the fact that using `visit` could potentially be expensive (with very little gained).